presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -95,6 +95,9 @@ public JoinNode(@JsonProperty("id") PlanNodeId id,
                 .build();
         checkArgument(inputSymbols.containsAll(outputSymbols), "Left and right join inputs do not contain all output symbols");
         checkArgument(!isCrossJoin() || inputSymbols.equals(outputSymbols), "Cross join does not support output symbols pruning or reordering");
+
+        checkArgument((!criteria.isEmpty()) || !leftHashSymbol.isPresent(), "Left hash symbol is only valid in an equijoin");
+        checkArgument((!criteria.isEmpty()) || !rightHashSymbol.isPresent(), "Right hash symbol is only valid in an equijoin");
```

Comment:
By this, I assume that you're asking for:
`checkArgument(!(criteria.isEmpty() && rightHashSymbols.isPresent()), ... );`
I will do that.